### ISSUE TITLE
axe tests don't run in Playwrights UI mode

### ISSUE THREAD

**Issue by AustinGil** [dequelabs/axe-core-npm#764]
### Product

playwright

### Product Version

1.35

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
In Playwright UI mode, the test assertions/expectations should be presented in the UI so you can see which tests fail. It would also be ideal if the failing tests could integrate with the visual highlighter so you can see which DOM node was responsible for the failing criteria.

#### Actual
As it is now, the plugin works fine in CLI mode and it reports all the issues. In UI mode, it doesn't really do anything. The only way I was able to sort of work with it was to run an analysis then `expect(violations).toEqual([])`. In this way, there is actually something to report and a way to see the failing issues. Otherwise, it doesn't report anything by default.

#### How to Reproduce
Install dependencies:

```bash
npm init playwright@latest && /
npm i -D @playwright/test @axe-core/playwright
```

Write the test that uses `@axe-core/playwright` to test the page.
```js
import { test, expect } from '@playwright/test';
import { AxeBuilder } from '@axe-core/playwright';

test.describe('homepage', () => {
  test('passes axe', async ({ page }) => {
    await page.goto('http://localhost:5173/');
    const axe = new AxeBuilder({ page })
    const { violations } = await axe.analyze();
    expect(violations).toEqual([]) // This is the best way I found to see actual issues, but it's not great.
  });
})
```

**Comment by straker** [comment]
I guess it depends on what you want from the output in Playwright UI. For tests, you're example is more or less how you would go about making sure axe passed (I would use `.toHaveLength(0)` myself). Even other libraries that use axe like jest-axe's `toHaveNoViolations` just checks that [the violations array is empty](https://github.com/NickColley/jest-axe/blob/main/index.js#L126). They have additional logs though to help with the messaging.

**Comment by AustinGil** [comment]
Yeah, I agree. The reason I use `.toEqual` is because that's the only way I was able to find that actually prints the failing violations somewhere. Unfortunately, it's as one big blob of things and it can be very noisy.

I used the Cypress plugin that works much nicer because each violation is called out individually and theres a way to see more details of individual violations in the console. I haven't found a good way to log things to the console in Playwright, however, so that may be a challenge.

**Comment by straker** [comment]
By the cypress plugin, do you mean cypress-axe? Looks like they also have to write [custom message logs](https://github.com/component-driven/cypress-axe/blob/master/src/index.ts#L109-L124) to show the violations in the Cypress UI in a nice format. 

In terms of having @axe-core/playwright support custom message logs to the Playwright UI, that's probably outside the scope of our package as our package is just a Playwright runner to run axe (and nothing more).

**Comment by AustinGil** [comment]
Yeah, that one. I know there is also a `playwright-axe` plugin too, but it also fails in UI mode. Yeah, I guess this highlights a bigger issue with Playwright where there may not be a way for developers to add feedback within the UI for things like this.

I guess there isn't much to be done with this ticket then. Feel free to leave it open for tracking, or close it. I'll see about reaching out to the Playwright team about allowing logs, and in that case maybe we can re-discuss adding some feedback in UI mode.

**Comment by padmavemulapati** [comment]
Verified with the latest canary version - `axe-core/playwright: ^4.7.4-alpha.409`
and     `@playwright/test: 1.36.0` and working fine, am seeing UI mode results

https://github.com/dequelabs/axe-core-npm/assets/47409579/62c130cb-179d-4108-85c1-219387229cae

## Environment:
| **Label**                         | **Value** |
| -------------------------- | ------ |
| **Product**                | _axe-core/playwright_ |
|**Version** | _4.7.4-alpha.409_ |
|**@playwright/test**|_1.36.0_|
| **OS-Details**      | _MAC- Intel Core i7 - 11.6.8 _ |
|**BrowserDetails**|_Chrome Version 114.0.5735.110 (Official Build) (64-bit)_|

**Comment by AustinGil** [comment]
@padmavemulapati it looks like that's the CLI mode. To run the test in UI mode, you need the `--ui` flag. Can you try `npx playwright test --ui`? See https://playwright.dev/docs/test-ui-mode
